.github/workflows/deploy-images-dockerhub.yml

```diff
@@ -0,0 +1,54 @@
+name: Deploy Images to DockerHub
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Tag to use'
+        required: true
+        type: string
+
+env:
+  GITHUB_WORKFLOW_FOLDER: ./.github/workflows
+  JEMPI_APP_PATH: ./JeMPI_Apps
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  prepare:
+    runs-on: ubuntu-22.04
+    steps:
+    - uses: actions/checkout@v4
+    - uses: ./.github/workflows/actions/prepare
+  build-save-deploy-images:
+    runs-on: ubuntu-22.04
+    needs: [prepare]
+    steps:
+    - uses: actions/checkout@v4
+    - id: validate-tag
+      run: |
+
+          CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+          if [ "$CURRENT_BRANCH" != "main" ] && [ "$CURRENT_BRANCH" != "master" ]; then
```

Comment:
You can rather use conditional steps in the yaml rather than in shell.

Reply:
Hi @rcrichton . Yes had thought of that, but the `if` expression doent allow you to specify the reason why the job was skipped. Using the run command gives some context in the logs in case someone is debugging

Reply:
Sure that's fine.